kogito-dmn-quickstart/pom.xml

```diff
@@ -0,0 +1,132 @@
+<?xml version="1.0"?>
+<project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns="http://maven.apache.org/POM/4.0.0"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+  <modelVersion>4.0.0</modelVersion>
+  <groupId>org.acme</groupId>
+  <artifactId>kogito-dmn-quickstart</artifactId>
+  <version>1.0.0-SNAPSHOT</version>
+  <properties>
+    <compiler-plugin.version>3.8.1</compiler-plugin.version>
+    <maven.compiler.parameters>true</maven.compiler.parameters>
+    <maven.compiler.source>11</maven.compiler.source>
+    <maven.compiler.target>11</maven.compiler.target>
+    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
+    <quarkus.platform.artifact-id>quarkus-universe-bom</quarkus.platform.artifact-id>
```

Comment:
we should use `quarkus-bom` artifact as in other quickstarts.

Reply:
?
We discussed this earlier in chat; why this comment?
I do not think so, see here:
https://github.com/quarkusio/quarkus-quickstarts/blob/3d5865045e72f261d3e98c19707b1da0f9361801/kogito-quickstart/pom.xml#L29
/cc @danielezonca

Reply:
Hm this seems completely against any guidance. I think we need to clarify the expectations here, cause I've noticed following comment in top-level README.md https://github.com/quarkusio/quarkus-quickstarts/tree/main#use-alternative-platforms - that's what led me to commenting in first place.

Moreover when talking with @aloubyansky and @mjurc it turns out that we're adding this PR in sort of a interim phase, between quarkus-platform BOMs are released and other repositories are adapted.
@aloubyansky @mjurc can you please advise on the best approach here?

Reply:
> Hm this seems completely against any guidance

please provide reference for this assertion.

Further, have you tried locally to do the command which is in all the guides, including this one?
Eg:
```
mvn io.quarkus:quarkus-maven-plugin:{quarkus-version}:create \
    -DprojectGroupId=org.acme \
    -DprojectArtifactId=kogito-dmn-quickstart \
    -Dextensions="dmn" \
    -DnoExamples
```

if you do so locally, you will notice the generated Quarkus application will adopt the `quarkus-universe-bom`.

So it seems a bit far fetched to me that you asserting my contribution goes [quote] _completely against any guidance_ [end quote].

Further, if you want me to change the Quickstart code, I kindly ask then to be instructed which command should I use to generate it instead; there is no point in the guide we ask User to do 1 thing, but then we would offer as a solution something _different_ from what would result by following the guide. Don't you agree?

Reply:
It has to be `io.quarkus.platform:quarkus-bom` and `io.quarkus.platform:quarkus-kogito-bom` going forward. The dev tools will be generating projects importing member-specific BOMs starting 2.1.0.Final scheduled to be released tomorrow. The universe BOM is not going to be promoted anymore, consider it to be deprecated.
By the dev tools I mean code.quarkus.io, the Maven plugin and the CLI.
If you run today `quarkus create -S 2.1`, it will generate a project importing the `io.quarkus.platform:quarkus-bom`, not the universe BOM.

Reply:
The BOM split should have been enabled in the currently recommended 2.0.3.Final but there was an issue that was in the way. Once 2.0.4.Final is released, the dev tools will be using the split BOM model for it as well.

Reply:
@aloubyansky to check I understand correctly.

If I wait ***tomorrow*** to run 
```
mvn io.quarkus:quarkus-maven-plugin:2.1.0.Final:create \
    -DprojectGroupId=org.acme \
    -DprojectArtifactId=kogito-dmn-quickstart \
    -Dextensions="dmn" \
    -DnoExamples
```

would it generate a pom.xml similar to this:

https://github.com/quarkusio/quarkus-quickstarts/blob/3d5865045e72f261d3e98c19707b1da0f9361801/kogito-quickstart/pom.xml#L20-L38

minor the xml comment of course.

Am I understanding correctly?

Reply:
Correct. To be precise, once the 2.1.0.Final is out (the schedule is https://github.com/quarkusio/quarkus/wiki/Release-Planning#21).
Once 2.0.4.Final is out, projects based on that version will also be generated using the new platform model.

Reply:
@tarilabs I didn't mean that statement as an assessment of the contribution. I meant to react to the comment we have inside the other pom file `Ideally, it should import the quarkus-universe-bom instead, so it doesn't have to import this kogito-bom`. I didn't mean to blame or block any contribution here, just want to make sure we do things in sync with recent changes. I am sorry for the misunderstanding.

Reply:
Cool, thanks @aloubyansky !
I guess I will need to await tomorrow anyway since I've just tried it but:
![image](https://user-images.githubusercontent.com/1699252/127316761-a8247706-532a-4268-a58b-51714e98bda1.png)
failed.

Thanks for having stated clearly: 
> The universe BOM is not going to be promoted anymore, consider it to be deprecated
in https://github.com/quarkusio/quarkus-quickstarts/pull/912#discussion_r678212154

@jstastny-cz
the information on this PR thread was not available to me, so I will wait tomorrow and regenerate the quickstart locally when `2.1.0.Final` is really out, and update as necessary this `pom.xml` in this PR then.

I guess we will need to remove this comment from this Quarkus repo then:
https://github.com/quarkusio/quarkus-quickstarts/blob/3d5865045e72f261d3e98c19707b1da0f9361801/kogito-quickstart/pom.xml#L29

for it to be removed.

Reply:
Oh, that's not good. You caught a moment when the registry is down. Help @gastaldi! It should work, it's just the generated project would be based on the currently recommended version, which is 2.0.3.Final.

If the registry was working you could executed this command to see the new platform model:
````
mvn io.quarkus:quarkus-maven-plugin:2.1.0.Final:create \
    -DprojectGroupId=org.acme \
    -DprojectArtifactId=kogito-dmn-quickstart \
    -Dextensions="dmn" \
    -DnoExamples \
    -DplatformArtifactId=quarkus-bom
    -DplatformVersion=2.1.0.CR1
````

Reply:
It looks like some infra issues caused the registry to go down. Can you please try again? I'm investigating what happened

Reply:
Using 

```
mvn io.quarkus:quarkus-maven-plugin:2.1.0.Final:create \
    -DprojectGroupId=org.acme \
    -DprojectArtifactId=kogito-dmn-quickstart \
    -Dextensions="dmn" \
    -DnoExamples \
    -DplatformArtifactId=quarkus-bom \
    -DplatformVersion=2.1.0.CR1
```
(second to last line using `\` 😄 )
worked locally 👍 

I will anyway wait for tomorrow or when either `2.1.0.Final` or `2.0.4.Final` is fully released, as I want to be 100% sure the command in the guide which is:

```
mvn io.quarkus:quarkus-maven-plugin:2.1.0.Final:create \
    -DprojectGroupId=org.acme \
    -DprojectArtifactId=kogito-dmn-quickstart \
    -Dextensions="dmn" \
    -DnoExamples
```

does indeed generate the expected output indeed with no other leftovers. Thank you for mentioning the possibilities as of today, it was helpful to have a sneak peek preview :) 👍